Slash rewrite Part 1

nostorian opened the pull request:
PR updates the original codebase with the following so far:

1. Completely deprecates prefix commands and switches over to the new slash ones
2. Replace `bash` dev command with `/eval` for now
3. Migrates all `developer` and `utility` commands to slash commands

sach-12 commented on src/cogs/interactions/slash/help.py:
```diff
@@ -0,0 +1,160 @@
+            .add_field(name="Uptime", value="`/uptime`", inline=False)
+            .add_field(name="Ping", value="`/ping`", inline=False)
+            .add_field(name="Support", value="`/support`", inline=False)
+            .set_footer(text="Page 1/3"),
```
is there a way to not hardcode the number of pages, but instead get them dynamically? take a look at this: https://github.com/sach-12/pesu-bot/blob/dev/src/cogs/interactions/slash/general.py

nostorian replied:
commit [070cfe3](https://github.com/nostorian/pesu-bot/commit/070cfe399f494d447dce57c3cdc065720f51cea2) resolves this issue.

sach-12 commented on src/cogs/interactions/slash/mongo.py:
```diff
@@ -0,0 +1,72 @@
+    def __init__(self, bot):
+        self.bot = bot
+
+    @app_commands.command(name="info", description="Get verification info about a user.")
```
let's rewrite this one. allow any verified person to run /info. return an embed with basic discord info about the user (username, profile pic, when they verified, and anything else you can think about). now if the person who ran the /info command is a mod/admin, send a follow-up ephermal message/embed with the privileged info, which is only PRN. lemme know if you need more clarification on this

sach-12 commented on src/requirements.txt:
```diff
@@ -1,2 +1,7 @@
-python-dotenv==1.0.0
-discord.py==2.3.2
+python-dotenv
```
always include versions in requirements. and can you briefly explain why the new packages?

nostorian commented on src/application.py:
```diff
@@ -19,16 +21,23 @@
-                cog = f"{root.replace('/', '.')}.{f[:-3]}"
-                await client.load_extension(cog)
-                logger.info(f"Loaded {cog}")
+    for path in Path("cogs").rglob("*.py"):
```
it was the first thing that came into my mind when i wasn't able to get os.walk to work, so refactored it to use path instead

nostorian commented:
> Really good PR so far. Great work! Left a few comments, take a look. I might leave more comments later. Apart from code-specific comments, some general comments:
> 
> * Can you merge it into dev instead of main?
> * You need to have a uniform naming convention. Stick to pascal and snake casing appropriately. What I would do is: variable names are always snake casing, class names are pascal casing. And if any constants exist (the whole config.json file), they should be all caps
> * Let's remove the eval command. We no longer use it anyway

I have switched the base branch to target `dev` branch instead and have replied to all the comments you have made so far. As for uniform naming convention, I will look into it in the later updates. And also I have removed the `/eval` command in the commit [0359883](https://github.com/nostorian/pesu-bot/commit/0359883d4fe7758d8ee5382e560de2a9d05191cf)

nostorian commented:
Ok so by the looks of it, the rewrite has been successfully completed and all commands have been properly transitioned to python leveraging slash commands 🎉(many thanks to @pmmdot as well for their efforts). Since  a DB schema update is coming soon, we can hold off on merging for now and then get back to it later.

nostorian commented:
@sach-12 
I have done the needful, please take a look and let me know if any further changes are required.

nostorian commented:
will reopen in a new branch from this repo directly once i am able to
